typescript/cli/ci-test.sh

```diff
@@ -0,0 +1,164 @@
+    --out /tmp \
+    --ism ./examples/multisig-ism.yaml \
+    --origin anvil2 --remotes anvil1 \
+    --key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 \
+    --yes
+
+CORE_ARTIFACTS_PATH=`find /tmp/core-deployment* -type f -exec ls -t1 {} + | head -1`
+echo "Core artifacts:"
+cat $CORE_ARTIFACTS_PATH
+
+AGENT_CONFIG_FILENAME=`ls -t1 /tmp | grep agent-config | head -1`
+
+echo "Deploying warp routes"
+yarn workspace @hyperlane-xyz/cli run hyperlane deploy warp \
+    --chains ./examples/anvil-chains.yaml \
+    --core $CORE_ARTIFACTS_PATH \
+    --config ./examples/warp-tokens.yaml \
+    --out /tmp \
+    --key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 \
+    --yes
+
+echo "Sending test message"
+yarn workspace @hyperlane-xyz/cli run hyperlane send message \
+    --origin anvil1 \
+    --destination anvil2 \
+    --chains ./examples/anvil-chains.yaml \
+    --core $CORE_ARTIFACTS_PATH \
+    --quick \
+    --key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 \
+    | tee /tmp/message1
+
+MESSAGE1_ID=`cat /tmp/message1 | grep "Message ID" | grep -E -o '0x[0-9a-f]+'`
+echo "Message 1 ID: $MESSAGE1_ID"
+
+WARP_ARTIFACTS_FILE=`find /tmp/warp-deployment* -type f -exec ls -t1 {} + | head -1`
+ANVIL1_ROUTER=`cat $WARP_ARTIFACTS_FILE | jq -r ".anvil1.router"`
+
+echo "Sending test warp transfer"
+yarn workspace @hyperlane-xyz/cli run hyperlane send transfer \
+    --origin anvil1 \
+    --destination anvil2 \
+    --chains ./examples/anvil-chains.yaml \
+    --core $CORE_ARTIFACTS_PATH \
+    --router $ANVIL1_ROUTER \
+    --type native \
+    --quick \
+    --key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 \
+    | tee /tmp/message2
+
+MESSAGE2_ID=`cat /tmp/message2 | grep "Message ID" | grep -E -o '0x[0-9a-f]+'`
+echo "Message 2 ID: $MESSAGE2_ID"
+
+if [[ $OSTYPE == 'darwin'* ]]; then
+    # Required because the -net=host driver only works on linux
+    DOCKER_CONNECTION_URL="http://host.docker.internal"
+else
+    DOCKER_CONNECTION_URL="http://127.0.0.1"
+fi
+
+for i in "anvil1 8545 ANVIL1" "anvil2 8555 ANVIL2"
+do
+    set -- $i
+    echo "Running validator on $1"
+    docker run \
+      --mount type=bind,source="/tmp",target=/data --net=host \
+      -e CONFIG_FILES=/data/${AGENT_CONFIG_FILENAME} -e HYP_VALIDATOR_ORIGINCHAINNAME=$1 \
+      -e HYP_VALIDATOR_REORGPERIOD=0 -e HYP_VALIDATOR_INTERVAL=1 \
+      -e HYP_BASE_CHAINS_${3}_CONNECTION_URL=${DOCKER_CONNECTION_URL}:${2} \
+      -e HYP_VALIDATOR_VALIDATOR_TYPE=hexKey \
+      -e HYP_VALIDATOR_VALIDATOR_KEY=0x2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6 \
+      -e HYP_VALIDATOR_CHECKPOINTSYNCER_TYPE=localStorage \
+      -e HYP_VALIDATOR_CHECKPOINTSYNCER_PATH=/data/${1}/validator \
+      -e HYP_BASE_TRACING_LEVEL=debug -e HYP_BASE_TRACING_FMT=compact \
+      gcr.io/abacus-labs-dev/hyperlane-agent:main ./validator > /tmp/${1}/validator-logs.txt &
+done
```

Comment:
I wonder if in future we can simplify this so that we don't require validators and relayers for the CLI testing
eg using the `CoreApp.process` via some `hyperlane relay` and maybe using a simpler ISM that doesn't rely on validator signatures

Reply:
I would **love** that